src/main/java/com/leahtcodes/Controllers/UserController.java

```diff
@@ -19,6 +28,27 @@ public UserController(UserRepository userRepository){
         this.userRepository = userRepository;
     }
 
+    @Autowired
+    private OAuth2AuthorizedClientService authorizedClientService;
+
+    @GetMapping("/loginSuccess")
+    public String getLoginInfo(Userss user, OAuth2AuthenticationToken authentication) {
+        OAuth2AuthorizedClient client = authorizedClientService
+                .loadAuthorizedClient(
+                        authentication.getAuthorizedClientRegistrationId(),
+                        authentication.getName());
+                        user.setUsername(authentication.getName()) ;
+        //...
+        Map<String, Object> attributes = authentication.getPrincipal().getAttributes();
```

Comment:
You'll notice this is repeated from the other controller, might be something you could separate into another class that you could call from both places.